tests/estimator/test_tracking.py

```diff
@@ -0,0 +1,378 @@
+                            resource_rep(ResourceTestAlg1, {"num_iter": 3}): 2,
+                            resource_rep(ResourceTestZ): 4,
+                        },
+                    ),
+                ),
+            ),
+            (
+                {"TestCNOT", "TestT", "TestHadamard"},
+                Resources(
+                    zeroed=8,
+                    any_state=2,
+                    algo_wires=4,
+                    gate_types=defaultdict(
+                        int,
+                        {
+                            resource_rep(ResourceTestT): 416,
+                            resource_rep(ResourceTestCNOT): 6,
+                            resource_rep(ResourceTestHadamard): 6,
+                        },
+                    ),
+                ),
+            ),
+        ),
+    )
+    def test_varying_gate_sets(self, gate_set, expected_resources):
+        """Test that changing the gate_set correctly updates the resources"""
+
+        def my_circ(num_wires):
+            ResourceTestAlg2(num_wires, wires=range(num_wires))
+            for w in range(num_wires):
+                ResourceTestZ(wires=w)
+
+        actual_resources = estimate(my_circ, gate_set=gate_set)(num_wires=4)
+        assert actual_resources == expected_resources
+
+    @pytest.mark.parametrize("error_val", (0.1, 0.01, 0.001))
+    def test_varying_config(self, error_val):
+        """Test that changing the resource_config correctly updates the resources"""
+        custom_config = ResourceConfig()
+        custom_config.resource_op_precisions[ResourceTestRZ] = {"precision": error_val}
+
+        op = ResourceTestRZ()  # don't specify precision
+        computed_resources = estimate(op, gate_set={"TestT"}, config=custom_config)
+
+        expected_resources = Resources(
+            zeroed=0,
+            algo_wires=1,
+            gate_types=defaultdict(int, {resource_rep(ResourceTestT): round(1 / error_val)}),
+        )
+
+        assert computed_resources == expected_resources
+
+    @pytest.mark.xfail(reason="RX and equivalent not yet in PL core")
+    @pytest.mark.parametrize("error_val", (0.1, 0.01, 0.001))
+    def test_varying_single_qubit_rotation_precision(self, error_val):
+        """Test that setting the single_qubit_rotation_precision correctly updates the resources"""
+        custom_config = ResourceConfig()
+        custom_config.set_single_qubit_rot_precision(error_val)
+
+        # TODO: uncomment this when ready resource operators are imported.
+        # custom_config.set_decomp(ResourceRX, mock_rotation_decomp)
+        # custom_config.set_decomp(ResourceRY, mock_rotation_decomp)
+        # custom_config.set_decomp(ResourceRZ, mock_rotation_decomp)
+
+        # def my_circuit():
+        #     ResourceRX(wires=0)
+        #     ResourceRY(wires=1)
+        #     ResourceRZ(wires=2)
+
+        # computed_resources = estimate(my_circuit, gate_set={"TestT"}, config=custom_config)()
```

Comment:
[optional] Might just remove them and add them back later.

Reply:
I removed it for now. We might need additional tests for code cov though.